linux_os/guide/system/permissions/restrictions/coredumps/disable_users_coredumps/bash/shared.sh

```diff
@@ -1,8 +1,10 @@
 # platform = Red Hat Virtualization 4,multi_platform_rhel,multi_platform_ol
 SECURITY_LIMITS_FILE="/etc/security/limits.conf"
 
-if grep -qE '\*\s+hard\s+core' $SECURITY_LIMITS_FILE; then
+if grep -qE '^\s*\*\s+hard\s+core' $SECURITY_LIMITS_FILE; then
         sed -ri 's/(hard\s+core\s+)[[:digit:]]+/\1 0/' $SECURITY_LIMITS_FILE
 else
         echo "*     hard   core    0" >> $SECURITY_LIMITS_FILE
 fi
+
+sudo sed -ri '/^\s*\*\s+hard\s+core/d' /etc/security/limits.d/*.conf
```

Comment:
Don't use `sudo` in remediation script. Users are expected to run them under sudo or as root anyway.